tools/flash_shelly.py

```diff
@@ -0,0 +1,499 @@
+    }
+  return(options[type])
+
+def write_flash(device, lfw, dlurl, cfw_type, mode):
+  logger.info("\n" + WHITE + "write_flash" + NC)
+  flashed = False
+  host = device.replace('.local','')
+  if cfw_type == 'homekit':
+    print("Downloading Firmware...")
+    logger.info('DURL: %s' % dlurl)
+    myfile = requests.get(dlurl)
+    open('shelly-flash.zip', 'wb').write(myfile.content)
+    if path.exists('shelly-flash.zip') or cfw_type == 'stock':
+      print("Now Flashing...")
+      files = {
+          'file': ('shelly-flash.zip', open('shelly-flash.zip', 'rb')),
+      }
+      response = requests.post('http://%s/update' % device , files=files)
+      logger.info(response.text)
+  else:
+    print("Now Flashing...")
+    dlurl = dlurl.replace('https', 'http')
+    logger.info("curl -qsS http://%s/ota?url=%s" % (device, dlurl))
+    response = requests.get('http://%s/ota?url=%s' % (device, dlurl))
+    logger.info(response.text)
+
+  sleep(2)
+  n = 1
+  waittextshown = False
+  info = None
+  while n < 40:
+    if waittextshown == False:
+      print("waiting for %s to reboot" % host)
+      waittextshown = True
+    if mode == 'homekit':
+      checkurl = 'http://%s/rpc/Shelly.GetInfo' % device
+    else:
+      checkurl = 'http://%s/Shelly.GetInfo' % device
+    try:
+      fp = urllib.request.urlopen(checkurl)
+      info = json.load(fp)
+      fp.close()
+    except (urllib.error.HTTPError, urllib.error.URLError) as err:
+      logger.info("Error: %s" % err)
+      n += 1
+    if info:
+      n=41
+    sleep(1)
+
+  if mode == 'homekit':
+    onlinecheck = info['version']
+  else:
+    onlinecheck = info['fw'].split('/v')[1].split('@')[0]
+
+  if onlinecheck == lfw:
+    print(GREEN + "Successfully flashed %s to %s\033[0m" % (host, lfw))
+    if mode == 'homekit':
+      remove('shelly-flash.zip')
+    else:
+      if info['type'] == "SHRGBW2":
+        print("\nTo finalise flash process you will need to switch 'Modes' in the device WebUI,")
+        print(WHITE + "WARNING!!" + NC + "If you are using this device in conjunction with Homebridge it will")
+        print("result in ALL scenes / automations to be removed within HomeKit.")
+        print("Goto http://$device in your web browser")
+        print("Goto settings section")
+        print("Goto 'Device Type' and switch modes")
+        print("Once mode has been changed, you can switch it back to your preferred mode.")
+  else:
+    print(RED + "Failed to flash %s to %s\033[0m" % (host, lfw))
+    print("Current: %s" % onlinecheck)
+
+def probe_info(device, action, dry_run, silent_run, mode, exclude, exclude_device, forced_version, ffw):
+  flash = False
+  info = None           # firmware versions info
+  model = None          # device model
+  lfw = None            # latest firmware available
+  cfw = None            # current firmware on device
+  cfw_type = 'homekit'  # current firmware type
+  host = device
+  host = host.replace('.local','')
```

Comment:
always put spaces after commas in argument lists

Reply:
done https://github.com/mongoose-os-apps/shelly-homekit/pull/176/commits/249ad1373b919b4a56bbb8355216dada8c2ce975